### PR TITLE
imp(proto): Add Protobufs and adjust scripts

### PR DESCRIPTION
**Pull request by MalteHerrmann**
This PR adds the Protobuf implementations for the evmOS modules to the repository. Along with copy-pasting the implementations, the corresponding module and package paths are adjusted.
The new module structure is now e.g. `os.feemarket.v1` instead of `ethermint.feemarket.v1`. This is to avoid duplication of the Evmos repository, which has e.g. `evmos.erc20.v1` as its module name.

In comparison to the Evmos repository, the Swagger stuff was removed (for now) and can be added or adjusted in a later PR.

## Summary by CodeRabbit

- **New Features**
	- Enhanced documentation with an updated changelog for improved clarity on recent project modifications, including Protobuf integration.
	- Introduction of new protocol buffer files to support ERC20 token management, EVM functionality, fee market handling, and dynamic fee transactions.
	- New gRPC services for querying ERC20 token pairs and fee market parameters.

- **Improvements**
	- Streamlined Protobuf generation process and a more integrated approach for managing dependencies.
	- Added various event messages to enhance tracking and management of ERC20 token operations and Ethereum transactions.

- **Bug Fixes**
	- Improved echo statements in scripts for better user feedback during execution.

- **Chores**
	- Automated generation of protocol buffer files through a new script, simplifying workflows.
	- Updated Makefile to reduce complexity and improve the clarity of the build process.

### CODE REVIEW

File: scripts/generate_protos.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+# --------------
+# Commands to run locally
+# docker run --network host --rm -v $(CURDIR):/workspace --workdir /workspace ghcr.io/cosmos/proto-builder:v0.11.6 sh ./protocgen.sh
```
Comment (MalteHerrmann) [inline scripts/generate_protos.sh:5]:
```suggestion
# docker run --network host --rm -v $(CURDIR):/workspace --workdir /workspace ghcr.io/cosmos/proto-builder:v0.11.6 sh ./generate_protos.sh
```

**Review by Vvaradinov — APPROVED** [review]
LGTM!

File: Makefile [hunk 2]
```diff
@@ -276,10 +215,15 @@ check-licenses:
 	@python3 check_licenses.py .
 	@rm check_licenses.py
 
-check-changelog:
+# Use changelog linter and auto-fix from https://github.com/MalteHerrmann/changelog-utils
```
Comment (0xstepit) [inline Makefile:218]: ❤️

File: proto/os/types/v1/indexer.proto [hunk 3]
```diff
@@ -0,0 +1,32 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
```
Comment (0xstepit) [inline proto/os/types/v1/indexer.proto:2]: I would create a new LICENSE file in this repo and point this link to that one tbh

**Review by 0xstepit — APPROVED** [review]
Great refactoring Malte! Left a minor comment

File: proto/os/types/v1/indexer.proto (hunk 3 above)
Comment (MalteHerrmann) [reply proto/os/types/v1/indexer.proto:2]: as discussed on Slack, will adjust on a separate PR for all files

**Pull request merged**